api/tokens.go

```diff
@@ -13,7 +13,11 @@ import (
 
 //TokensHandler uses submitted email address and password to sign a user in against Cognito and returns a http handler interface
 func (api *API) TokensHandler(ctx context.Context, w http.ResponseWriter, req *http.Request) (*models.SuccessResponse, *models.ErrorResponse) {
-	defer req.Body.Close()
+	defer func() {
+		if err := req.Body.Close(); err != nil {
+			err = models.NewError(ctx, err, models.BodyCloseError, models.BodyClosedFailedDescription)
```

Comment:
err is not being logged, maybe replace this line with:
log.Event(ctx, models.BodyClosedFailedDescription, log.Error(err), log.ERROR)

Reply:
The models.NewError function logs the error. This is too make sure all errors are logged when they have been generated in to the required struct.
[https://github.com/ONSdigital/dp-identity-api/blob/develop/models/errors.go#L23](https://github.com/ONSdigital/dp-identity-api/blob/develop/models/errors.go#L23)

Reply:
Ah thanks.
Then in which case, the line with:
err = models.NewError(ctx, err, models.BodyCloseError, models.BodyClosedFailedDescription)
should really be:
_ = models.NewError(ctx, err, models.BodyCloseError, models.BodyClosedFailedDescription)

to clearly show that code is not interested in return value from NewError ?

Reply:
Yeah thats a good point.
I've updated now.